shell/platform/fuchsia/flutter/tests/session_connection_unittests.cc

```diff
@@ -34,9 +34,19 @@ class SessionConnectionTest : public ::testing::Test {
     view_token_ = std::move(view_token);
 
     scenic_->CreateSession(session_.NewRequest(), session_listener_.Bind());
-    presenter_->PresentView(std::move(view_holder_token), nullptr);
+    presenter_->PresentOrReplaceView(std::move(view_holder_token), nullptr);
 
     FML_CHECK(zx::event::create(0, &vsync_event_) == ZX_OK);
+
+    // Ensure Scenic has had time to wake up before the test logic begins.
```

Comment:
In tree we can use a RealLoopFixture that allows us things like RunLoopUntil() which functionally does what I've recreated here. We can also do hermetic testing in tree which means we don't have to spin up a real Scenic that connects to the real display, etc. We couldn't export that outside of the tree either - we looked into it back in December when I first added these tests if you remember.

Filed: https://github.com/flutter/flutter/issues/61768

Reply:
Ah yeah, I remember that now.  The out-of-tree testing situation is so difficult!

Thanks for filing the bug

Reply:
Is there an issue filed somewhere for improved out-of-tree testing support? If I'm understanding correctly, it sounds like that is the core issue here.

Reply:
In a perfect world we could use Fuchsia's TestWithEnvironment to make these tests hermetic and match in-tree behavior, but TestWithEnvironment is not exported to the SDK. I can copy and paste part of the conversation here about the difficulties surrounding that effort:

Adam Barth:

"Yes, notice that TestWithEnvironment depends on gtest:

class TestWithEnvironment : public gtest::RealLoopFixture

The Flutter tree might have a different version of gtest than the Fuchsia tree, which means we cannot include code in the Fuchsia SDK that depends on gtest.  Even more problematically, gtest itself depends on the Fuchsia SDK, which means introducing a dependency from the Fuchsia SDK to gtest would create a circular dependency.

Instead, I'd recommend creating your own version of this code in the Flutter tree that uses the Flutter instance of gtest."

Jaeheon Yi:

"Do keep in mind that forked libraries will require some ongoing maintenance.  Today there's no discrepancy between these two versions, because the forking is fresh.  What happens in a year, after features/bugfixes have been added to the Fuchsia version, or when a Components v2-specific "TestWithEnvrionment" gets authored?"

Reply:
Do we have any sense for the amount of work required for abarth's suggestion "Create your own version of this code in the Flutter tree"? @iskakaushik @gw280

Reply:
Note that *TestWithEnvironment itself* cannot be exposed in the SDK, but that is just the test fixture class.

The vast, vast majority of the environment-faking code that test fixture uses does not have any problematic dependencies.  Exposing that in the SDK would be quite easy to do, and then only the test fixture itself needs forking.  We can make the fixture code itself extremely small so that isn't a problem.

Reply:
Another possibility that came from a conversation this morning:

We could construct blackbox TestWithEnvironment stuff like this in the Fuchisa tree, and run it in fuchsia's CQ.  Those tests should not be reaching into Flutter's internals anyways so the lack of flutter source is actually a plus.

For unit tests like this one, maybe we can use CMX injection via facets to fake stuff out